indexer/Driver.cc

```diff
@@ -294,19 +297,19 @@ class Driver {
 
   void assignJobToWorker(WorkerId workerId, JobId jobId) {
     // TODO(ref: add-job-debug-helper) Print abbreviated job data here.
-    spdlog::info("assigning jobId {} to worker {}", jobId.id(), workerId);
+    spdlog::debug("assigning jobId {} to worker {}", jobId.id(), workerId);
     this->wipJobs.insert(jobId);
     this->markWorkerBusy(workerId, jobId);
     auto it = this->allJobList.find(jobId);
     ENFORCE(it != this->allJobList.end(), "trying to assign unknown job");
     this->queues.driverToWorker[workerId].send(
-        IndexJobRequest{it->first, it->second});
+        IndexJobRequest{it->first, std::move(it->second)});
+    this->allJobList.erase(it);
```

Comment:
TODO: We shouldn't erase the job here, we just need to change the status. We may need to perform an explicit copy instead of a move though.